### PR TITLE
[CMake] Look for Python3 again in `SearchInstalledSoftware.cmake`

### PR DESCRIPTION
**Pull request by guitargeek**
This commit essentially reverts a288ff9 from four years ago, which was also part of the multi-python machinery that is not necessary anymore after Python 2 support was dropped.

Also, make the choice of looked-up components consistent with what is actually required.

### CODE REVIEW

**Review by dpiparo — APPROVED** [review]
Thanks for these changes, LGTM, but please could you address my comments?

File: CMakeLists.txt [hunk 1]
```diff
@@ -112,7 +112,6 @@ set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules)
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 
 #---Load some basic macros which are needed later for the confiuration and build----------------
-include(SearchRootCoreDeps)
```
[the remark this answers is not in the document]
Comment (guitargeek) [reply CMakeLists.txt:115]: The file doesn't exist anymore. It contained some of the multi-python (2 and 3) logic that is gone now

File: interpreter/CMakeLists.txt [hunk 2]
```diff
@@ -372,10 +372,10 @@ else()
       message(FATAL_ERROR
 "Unable to find Python interpreter, required for builds and testing.
 
-Please install Python or specify the PYTHON_EXECUTABLE CMake variable.")
+Please install Python or specify the Python3_EXECUTABLE CMake variable.")
     endif()
 
-    if( ${PYTHON_VERSION_STRING} VERSION_LESS 2.7 )
+    if( ${Python3_VERSION} VERSION_LESS 2.7 )
```
[the remark this answers is not in the document]
Comment (guitargeek) [reply interpreter/CMakeLists.txt:378]: Ah yes, this is indeed impossible! The check should be removed. We don't support Python 2 anyway

**Comment by guitargeek** [comment]
Thanks a lot for the review! Can you maybe review the corresponding PR in roottest too? Thanks! https://github.com/root-project/roottest/pull/1120

**Pull request merged**